src/app/clusters/scenes-server/SceneHandlerImpl.cpp

```diff
@@ -16,10 +16,156 @@
+        Type minVal = ConvertByteArrayToType<Type>(minMaxValue->minValue, emberAfAttributeSize(metadata));
+        Type maxVal = ConvertByteArrayToType<Type>(minMaxValue->maxValue, emberAfAttributeSize(metadata));
+
+        // Cap based on minimum value
+        if (minVal > static_cast<Type>(aVPair.attributeValue))
+        {
+            uint64_t sValue = 0;
+            memcpy(&sValue, &minVal, sizeof(Type));
+            aVPair.attributeValue = app::NumericAttributeTraits<uint64_t>::StorageToWorking(sValue);
+            // We assume the max is >= min therefore we can return
+            return;
+        }
+
+        // Adjust maxValue if greater than the meta data's max value
+        if (maxVal < maxValue)
+        {
+            maxValue = maxVal;
+        }
+    }
+
+    // Cap based on maximum value
+    if (metadata->IsSignedIntegerAttribute())
+    {
+        if (static_cast<int64_t>(aVPair.attributeValue) > static_cast<int64_t>(maxValue))
+        {
+            uint64_t sValue = 0;
+            memcpy(&sValue, &maxValue, sizeof(Type));
+            aVPair.attributeValue = app::NumericAttributeTraits<uint64_t>::StorageToWorking(sValue);
+        }
+    }
+    else
+    {
+        if (static_cast<uint64_t>(aVPair.attributeValue) > static_cast<uint64_t>(maxValue))
+        {
+            uint64_t sValue = 0;
+            memcpy(&sValue, &maxValue, sizeof(Type));
+            aVPair.attributeValue = app::NumericAttributeTraits<uint64_t>::StorageToWorking(sValue);
+        }
+    }
+}
+
+/// @brief  Validate the attribute exists for a given cluster
+/// @param[in] endpoint   Endpoint ID
+/// @param[in] clusterID  Cluster ID
+/// @param[in] aVPair     AttributeValuePairType, will be mutated to cap the value if it is out of range
+/// @return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute) if the attribute does not exist for a given cluster or is not scenable
+/// @note This will allways fail for global list attributes. If we do want to make them scenable someday, we will need to
+///       use a different validation method.
+// TODO: Add check for "S" quality to determine if the attribute is scenable once suported :
+// https://github.com/project-chip/connectedhomeip/issues/24177
+CHIP_ERROR ValidateAttributePath(EndpointId endpoint, ClusterId cluster, AttributeValuePairType & aVPair)
+{
+    const EmberAfAttributeMetadata * metadata = emberAfLocateAttributeMetadata(endpoint, cluster, aVPair.attributeID);
+
+    if (nullptr == metadata)
+    {
+        return CHIP_IM_GLOBAL_STATUS(UnsupportedAttribute);
+    }
+
+    switch (metadata->attributeType)
+    {
+    case ZCL_INT8U_ATTRIBUTE_TYPE:
+        CapAttributeID<uint8_t>(aVPair, metadata);
+        break;
+    case ZCL_INT16U_ATTRIBUTE_TYPE:
+        CapAttributeID<uint16_t>(aVPair, metadata);
+        break;
+    case ZCL_INT32U_ATTRIBUTE_TYPE:
+        CapAttributeID<uint32_t>(aVPair, metadata);
+        break;
+    case ZCL_INT64U_ATTRIBUTE_TYPE:
```

Comment:
What happened to 24, 40, 48, 56?  We absolutely have int24* attributes around in standardized clusters.  Incidentally, that's why `NumericAttributeTraits` comes in handy; together with `OddSizedInteger` it helps you handle all that stuff sanely (e.g. `NumericAttributeTraits<OddSizedInteger<3, true>>::StorageToWorking` will convert from a 3-byte buffer to a 32-bit unsigned integer that you can then actually work with....

Reply:
So the Scenes Spec doesn't allow those types but does say in the event we would ever add a "Scene" quality attribute with an odd sized integer, we should convert it to one of the types in this case.